vic/drivers/cesm/src/init_library.c

```diff
@@ -0,0 +1,251 @@
+        x2l_vic[i].x2l_Faxa_rainl = MISSING;
+        x2l_vic[i].x2l_Faxa_snowc = MISSING;
+        x2l_vic[i].x2l_Faxa_snowl = MISSING;
+        x2l_vic[i].x2l_Faxa_swndr = MISSING;
+        x2l_vic[i].x2l_Faxa_swvdr = MISSING;
+        x2l_vic[i].x2l_Faxa_swndf = MISSING;
+        x2l_vic[i].x2l_Faxa_swvdf = MISSING;
+        x2l_vic[i].x2l_Sa_co2prog = MISSING;
+        x2l_vic[i].x2l_Sa_co2diag = MISSING;
+        x2l_vic[i].x2l_Faxa_bcphidry = MISSING;
+        x2l_vic[i].x2l_Faxa_bcphodry = MISSING;
+        x2l_vic[i].x2l_Faxa_bcphiwet = MISSING;
+        x2l_vic[i].x2l_Faxa_ocphidry = MISSING;
+        x2l_vic[i].x2l_Faxa_ocphodry = MISSING;
+        x2l_vic[i].x2l_Faxa_ocphiwet = MISSING;
+        x2l_vic[i].x2l_Faxa_dstwet1 = MISSING;
+        x2l_vic[i].x2l_Faxa_dstwet2 = MISSING;
+        x2l_vic[i].x2l_Faxa_dstwet3 = MISSING;
+        x2l_vic[i].x2l_Faxa_dstwet4 = MISSING;
+        x2l_vic[i].x2l_Faxa_dstdry1 = MISSING;
+        x2l_vic[i].x2l_Faxa_dstdry2 = MISSING;
+        x2l_vic[i].x2l_Faxa_dstdry3 = MISSING;
+        x2l_vic[i].x2l_Faxa_dstdry4 = MISSING;
+        x2l_vic[i].x2l_Flrr_flood = MISSING;
+        x2l_vic[i].x2l_vars_set = false;
+    }
+}
+
+/******************************************************************************
+ * @brief    Initialize l2x_data_struct.
+ *****************************************************************************/
+void
+initialize_l2x_data()
+{
+    extern l2x_data_struct *l2x_vic;
+    extern domain_struct    local_domain;
+
+    size_t                  i;
+
+    log_info("Setting all l2x fields to %f", MISSING);
+
+    for (i = 0; i < local_domain.ncells; i++) {
+        l2x_vic[i].l2x_Sl_t = MISSING;
+        l2x_vic[i].l2x_Sl_tref = MISSING;
+        l2x_vic[i].l2x_Sl_qref = MISSING;
+        l2x_vic[i].l2x_Sl_avsdr = MISSING;
+        l2x_vic[i].l2x_Sl_anidr = MISSING;
+        l2x_vic[i].l2x_Sl_avsdf = MISSING;
+        l2x_vic[i].l2x_Sl_anidf = MISSING;
+        l2x_vic[i].l2x_Sl_snowh = MISSING;
+        l2x_vic[i].l2x_Sl_u10 = MISSING;
+        l2x_vic[i].l2x_Sl_ddvel = MISSING;
+        l2x_vic[i].l2x_Sl_fv = MISSING;
+        l2x_vic[i].l2x_Sl_ram1 = MISSING;
+        l2x_vic[i].l2x_Sl_soilw = MISSING;
+        l2x_vic[i].l2x_Sl_logz0 = MISSING;
+        l2x_vic[i].l2x_Fall_taux = MISSING;
+        l2x_vic[i].l2x_Fall_tauy = MISSING;
+        l2x_vic[i].l2x_Fall_lat = MISSING;
+        l2x_vic[i].l2x_Fall_sen = MISSING;
+        l2x_vic[i].l2x_Fall_lwup = MISSING;
+        l2x_vic[i].l2x_Fall_evap = MISSING;
+        l2x_vic[i].l2x_Fall_swnet = MISSING;
+        l2x_vic[i].l2x_Fall_fco2_lnd = MISSING;
+        l2x_vic[i].l2x_Fall_flxdst1 = MISSING;
+        l2x_vic[i].l2x_Fall_flxdst2 = MISSING;
+        l2x_vic[i].l2x_Fall_flxdst3 = MISSING;
+        l2x_vic[i].l2x_Fall_flxdst4 = MISSING;
+        l2x_vic[i].l2x_Fall_flxvoc = MISSING;
+        l2x_vic[i].l2x_Flrl_rofliq = MISSING;
+        l2x_vic[i].l2x_Flrl_rofice = MISSING;
+        l2x_vic[i].l2x_vars_set = false;
```

Comment:
@apcraig - Do you think it is better to return a missing value or zero for `l2x` fields that we are not using?  Is there a convention for this?